### PR TITLE
Remove related pipeline logs during pipeline deletion

### PR DESCRIPTION
**Pull request by xoxys**
This PR ensures related pipeline logs are deleted as well during pipeline deletion. To clean up orphaned log entries, I have used the following query on my postgres db:

```sql
DELETE FROM log_entries
USING (
    SELECT l.id 
    FROM log_entries l
    LEFT JOIN steps s ON l.step_id = s.id
    WHERE s.id IS NULL
) AS orphaned
WHERE log_entries.id = orphaned.id;
```

**Disclaimer:** I'm not responsible for any damage to your database, use the query at your own risk and make a backup before executing it.

### CODE REVIEW

**Comment by zc-devs** [comment]
> To clean up orphaned log entries, I have used the following query on my postgres db

Shouldn't it be done by migration?

**Comment by xoxys** [comment]
I would say no as it can be really (really) time-consuming on huge tables and would delay the server start massively. IMO adding a cli command to `woodpecker-server` like gitea has it with `gitea doctor` would be a better choice.

**Comment by zc-devs** [comment]
Thanks for clarification.

This is a bug. Bug in what feature? Could you link it here? If this feature was merged in `next` (not released yet, or RC counts?), then maybe there is no need in migration or tool.

**Comment by xoxys** [comment]
The pipelineDelete API exists for a long time already, even in the last releases. So users that have used it already will have orphaned log entries in the db, and that's what I call a bug. Im going to mark this discussion as offtopic.

**Comment by zc-devs** [comment]
How/why users could/can delete pipeline without deleting related workflow entities?
How/why users could/can delete workflow without deleting related steps entities?
How/why users could/can delete step without deleting related logs entities?
Isn't there a bug in the database schema?
Shouldn't an option like `cascade` be in API instead of hardcoded behavior?

**Comment by xoxys** [comment]
> Can you put that logic into `deleteStep`?

How? DeleteStep does not have the storage client.

**Comment by xoxys** [comment]
As I said the linked `deleteStep` method does not have access to the storage client. I have refactored it a bit. What do you think?

File: server/store/datastore/step.go [hunk 1]
```diff
@@ -84,6 +84,9 @@ func (s storage) StepUpdate(step *model.Step) error {
 }
 
 func deleteStep(sess *xorm.Session, stepID int64) error {
+	if err := logDelete(sess, stepID); err != nil {
+		return err
+	}
 	if _, err := sess.Where("id = ?", stepID).Delete(new(model.LogEntry)); err != nil {
```
[the remark this answers is not in the document]
Comment (xoxys) [reply server/store/datastore/step.go:90]: Wtf... ok now things got clearer :rofl:

[the remark this answers is not in the document]
Comment (xoxys) [reply server/store/datastore/step.go]: Done

**Comment by pat-s** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|release/v2.8|[](https://github.com/woodpecker-ci/woodpecker/pull/4694)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

**Pull request merged**